File: IE 211 Fall21/src/Lab1/Animal.java
package Lab1;

abstract class Animal {

	private String name;
	private int id;

	public Animal() {}
	
	public void setName(String nameIn)
	{
		name = nameIn;
	}
	
	public void setID(int idIn)
	{
		id = idIn;
	}
	
	public int getIDLastDigit()
	{
		return id % 10;
	}
	
	public abstract void speak();
	
}

File: IE211 Fall22/src/Lab1/Animal.py
class Animal:
    __me = {
        "name": None,
        "id": 0
    }

    def __init__(self):
        pass

    def setName(self, name):
        self.__me["name"] = name

    def setID(self, id):
        self.__me["id"] = id

    def getIDLastDigit(self):
        return self.__me["id"] % 10

    def speak(self):
        print("Hi all :)")


